### ISSUE TITLE
telethon client.connect() did not response

### ISSUE THREAD

**Issue by nikraftar** [LonamiWebs/Telethon#496]
hi.
i use telethon for windows 7 python 3
this is my code to connect

from telethon import TelegramClient
api_id = ?????
api_hash = '???????'
phone = '+????'
client = TelegramClient('session_name', api_id, api_hash)
print("start...")
client.connect()
print("finished")

the start... will print then nothing happened. no error no code nothing
"session_name.session" file created too
what am i doing wrong?

**Comment by Lonami** [comment]
Not sure. Bad network conditions? Enable logging? Where does it halt?

**Comment by nikraftar** [comment]
network is fine. what mean by enable logging?
in client.connect() line program did not any response or halt
i execute this code in telethon_generator folder

**Comment by Lonami** [comment]
> i execute this code in telethon_generator folder

What?

**Comment by Lonami** [comment]
Well… I can't see why you would run that code *there*. Just install with `pip` use in your own folder…

**Comment by nikraftar** [comment]
i did it. still same issue. stuck in client.connect() line :(

**Comment by nikraftar** [comment]
after some minutes i got this error
`ValueError: Number of retries reached 0`

**Comment by zeroryuki** [comment]
im having that exact problem as well. 2 days in a row. no error, no response, nothing at all.. just stuck at there

**Comment by TgWindoz** [comment]
same error

**Comment by TgWindoz** [comment]
on windows 7 it works on raspbian 9 no

**Comment by hrmele** [comment]
same here. running on ubuntu.
used to work until a month ago.
went to use today, and got stuck.

deleted the auth file. nothing happens.
I call "client.connect()" and it stays stuck forever.

**Comment by Lonami** [comment]
It would be useful if you enabled logging…

**Comment by hrmele** [comment]
could you help me to do that? I am not that experienced with python, but if you give me some guidance I can try to troubleshot this problem.

**Comment by Lonami** [comment]
```python
import logging
logging.basicConfig(level=logging.DEBUG)
```

**Comment by hrmele** [comment]
DEBUG:telethon.network.mtproto_sender:Handling bad message notification
DEBUG:telethon.network.mtproto_sender:Read Bad Message error: (BadMessageError(...), 'msg_id too high (similar to the previous case, the client time has to be synchronized, and the message re-sent with the correct msg_id).')
DEBUG:telethon.network.mtproto_sender:Attempting to use the correct time offset.

and it keeps showing this message in loop

**Comment by Lonami** [comment]
1. Make sure you're using a recent version.
2. Make sure system time is correct…

**Comment by hrmele** [comment]
1) Updated to last varsion (was using 0.15)
2) Time is correct. Now the error message is different.

INFO: Connecting to Telegram Servers...  True
INFO:telethon.telegram_bare_client:Connecting to 149.154.167.51:443...
INFO:telethon.telegram_bare_client:Connection success!
DEBUG:telethon.telegram_bare_client:Invoking GetStateRequest
INFO:telethon.telegram_bare_client:Initializing a new connection while invoking
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: InvokeWithLayerRequest: 6526992913422920584
DEBUG:telethon.network.mtproto_sender:Processing BadServerSalt result
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: InvokeWithLayerRequest: 6526992914303178188
DEBUG:telethon.network.mtproto_sender:Processing BadMsgNotification result
WARNING:telethon.network.mtproto_sender:Read bad msg notification BadMsgNotification(bad_msg_seqno=4, bad_msg_id=6526992914304251072, error_code=17): msg_id too high (similar to the previous case, the client time has to be synchronized, and the message re-sent with the correct msg_id).
INFO:telethon.network.mtproto_sender:Attempting to use the correct time offset

**Comment by Lonami** [comment]
> Time is correct

Are you sure? Error reads "the client time has to be synchronized, […]".

**Comment by hrmele** [comment]
Yes, I know. It is weird, but time is right. First thing I checked when I saw this debug error messages.
You can see on the image. I am in Brazil, my timezone is GMT-3.

![image](https://user-images.githubusercontent.com/22687500/36699457-6e923dd2-1b2b-11e8-9a8c-a4f8e5744873.png)

**Comment by Lonami** [comment]
Well not sure I guess I can look into this *yet* again, the library should have tried its best to set the right timedelta…

**Comment by hrmele** [comment]
If there is anything I can do to help, just let me know.

**Comment by hrmele** [comment]
Here is a few things that may help 
https://github.com/sochix/TLSharp/issues/405

**Comment by hrmele** [comment]
not sure if that helps, but if I delete the session file, I can run the code normally.
I enter my phone number, DEBUG shows this error one time, but then I get the auth code on my phone, enter it on the program and it starts to work normally.

But if I end the program and try to start again, it gives me that error over and over.

**Comment by Lonami** [comment]
Regarding that link, I already attempt to set the proper time offset:
https://github.com/LonamiWebs/Telethon/blob/0b662f3b043907e872e49931201dc0fd923c63dd/telethon/session.py#L341-L345

When such error happens:
https://github.com/LonamiWebs/Telethon/blob/c83638ed0e168985f8edefd2cb2b1c266a288174/telethon/network/mtproto_sender.py#L393-L401

And I have modified that a few times now because the "bug" has been around for a while. I thought I was already doing the right thing but I guess I'll just look into it again.

**Comment by Lonami** [comment]
tl;dr; I was calculating and using the time offset right but there was an additional check that reused the incorrect ID.

**Comment by hrmele** [comment]
amazing
could you help me on how do I get the updated code to end this issue?
sorry for these dumb question. I am an old programmer not used to python and github.

**Comment by Lonami** [comment]
`pip3 install -U telethon`.
